sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -199,6 +199,13 @@ public void finish(
     this.finishStatus = FinishStatus.finishing(status);
     if (!root.isFinished()
         && (!transactionOptions.isWaitForChildren() || hasAllChildrenFinished())) {
+
+      // any un-finished childs will remain unfinished
+      // as relay takes care of setting the end-timestamp + deadline_exceeded
+      // see
+      // https://github.com/getsentry/relay/blob/40697d0a1c54e5e7ad8d183fc7f9543b94fe3839/relay-general/src/store/transactions/processor.rs#L374-L378
+      root.finish(finishStatus.spanStatus, finishTimestamp);
```

Comment:
@stefanosiano I basically moved this up, ensuring a finish date is set. I didn't see any negative side effects, hope that's ok!